sense_hat_example/sense_stick.js

```diff
@@ -0,0 +1,385 @@
+            // change the led
+            var ledIndex = data;
+            var ledData = window.sense_hat.pixels[ledIndex];
+            
+            var led = document.getElementById('e' + ledIndex);
+            
+            if (ledData[0] > lowlightLimit || ledData[1] > lowlightLimit || ledData[2] > lowlightLimit) {
+                var colorVal = "rgb(" + ledData[0] + "," + ledData[1] + "," + ledData[2] + ")";
+                led.style.fill = colorVal;
+            } else {
+                led.style.fill = "#F6F6F6";
+            }
+        } else if (event && event === 'changeLowlight') {
+            if (data === true) {
+                lowlightLimit = 8;
+            } else {
+                lowlightLimit = 47;
+            }
+        } else if (event && event === 'init') {
+            for (var i = 0; i < 64; i++) {
+                var ledData = window.sense_hat.pixels[i];
+                var led = document.getElementById('e' + i);
+                led.style.fill = "#F6F6F6";
+                window.sense_hat.pixels[i] = [0, 0, 0]; // reset
+            }
+        }
+    }
+    
+    var runbtn = document.getElementById('runbtn');
+    var stopbtn = document.getElementById('stopbtn');
+    var output = document.getElementById('output');
+
+    var interrupt = false;
+    var lowlightLimit = 47;
+    
+    function interruptHandler() {
+        if (interrupt === true) {
+            var newText = output.value + 'Stopping...';
+            output.value = newText;
+            throw new Error('Stopped the hard way!');
+        } else {
+            return null;
+        }
+    }
+
+    function cleanAfterRun() {
+        Sk.sense_hat.sensestick.destroy();
+
+        document.body.removeEventListener('keydown', handleKeyDown);
+        document.body.removeEventListener('keypress', handleKeyPress);
+        document.body.removeEventListener('keyup', handleKeyUp);
+    }
+
+    stopbtn.addEventListener('click', function (e) {
+        interrupt = true;
+
+        // first raise signal
+        sendSignalToSkulpt('SIGTERM', {});
+
+        window.sense_hat.sensestick.triggerKeyboardInterrupt(); // Otherwise we need to wait until the suspension resolves!
+    });
+    
+    runbtn.addEventListener('click', function (e) {
+        interrupt = false; // reset
+        output.value = ""; // reset output window
+        
+        function builtinRead(x) {
+            if (Sk.builtinFiles === undefined || Sk.builtinFiles["files"][x] === undefined)
+                throw "File not found: '" + x + "'";
+            return Sk.builtinFiles["files"][x];
+        }
+        (Sk.TurtleGraphics || (Sk.TurtleGraphics = {})).target = 'mycanvas';
+
+        Sk.configure({ 
+            read: builtinRead,
+            output: function (val) {
+                var newText = output.value + val;
+                output.value = newText;
+            },
+            killableWhile: true,
+            signals: true
```

Comment:
Signals requires the `signals: true` setting, otherwise signal.pause() resumes immediatelly.